### PR TITLE
Necessary evils/model controller specs

### PR DESCRIPTION
**Pull request by wlteo**
Tightened up model validations (tweets must be 1-140 chars, usernames must be at least 6 chars) and rewrote the model test specs accordingly.

All controller actions have test specs now.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
   it { should validate_presence_of(:email) }
   it { should validate_presence_of(:password) }
-  it { should have_many(:tweets) }
+  it do
+    should validate_length_of(:username).
+    is_at_least(6)
+  end
+
+  describe 'validates unqiueness of' do
+    subject { User.new(email: 'email@example.com', password: '123456', username: 'myusername')}
```
Comment (gohdaniel15) [inline spec/models/user_spec.rb]: You should use your user factory here too 😄

**Comment by gohdaniel15** [comment]
This code is long overdue. Lets make the necessary changes, or put them in another PR. Whatever it is, lets not leave our PR out there for too long. 

If you guys arent sure whether to merge this or not, label the PR as `WIP`. Else 🚀

File: spec/models/user_spec.rb (hunk 1 above)
Comment (wlteo) [reply spec/models/user_spec.rb]: Thanks for the tip! Just updated it to use factory-girl.

**Review by lerhhl — APPROVED** [review]
:rocket:

**Review by sabrina-s — APPROVED** [review]
LGTM! 🥇

**Pull request merged**